app_utils/config.py

```diff
@@ -45,8 +45,8 @@ def get_size(x):
     "allowed_file_extensions": os.getenv(
         "ALLOWED_FILE_EXTENSIONS", ".zip,.csv,.pq"
     ).split(","),
-    "data_folder": "data/",
-    "output_folder": "output/",
+    "data_folder": f"{os.getenv('DATA_FOLDER', 'data/')}",
+    "output_folder": f"{os.getenv('OUTPUT_FOLDER', 'output/')}",
```

Comment:
Definitely agree @pascal-pfeiffer . Any suggestions?

For example:

```suggestion
    "output_folder": f"{os.getenv('LLM_STUDIO_OUTPUT_FOLDER', 'output/')}",
```

Reply:
```python
"output_folder": str(os.getenv("LLM_STUDIO_OUTPUT_FOLDER", os. getcwd())) + "/output/"
```

Name is good. What about something along those lines? It should allow to use a single volume for both.

I think the Dockerfile will also need to be touched, though.

Reply:
This would enforce the end directory is always named exactly “output”. Works for the Docker image (/workspace/output) and my use case — single volume with 2 folders within — although would mean users are unable to change the name from “output” if they needed to. 

My preference is to keep it simple and flexible by letting the entire value be able to be overwritten. Although this works for me 👍 

+1 for defaulting to cwd. 

How about: 

```suggestion
"output_folder": str(os.getenv("LLM_STUDIO_OUTPUT_FOLDER", f”{os.getcwd()}/output/"))
```

?

Reply:
I also like to have it as flexible as possible. The issue is, that people may use the same ENV variable for both folders and that may lead to very unexpected results (e.g. experiment being named `user`). I think my proposal would only create one more folder nesting and as you said, enforce end directories to be named `output` and `data`.

Ultimately, I would probably adapt the basic Docker approach to use a single mounted volume.
So, we can replace these two lines with a single volume that is attached to the Docker image

```
    -v `pwd`/data:/workspace/data \
    -v `pwd`/output:/workspace/output \
```

with

```
    -v `pwd`:/workspace\
```

while also allowing to pass the ENV var you added to the Docker image and customizing the folders individually for very specific use cases where the two folders need to sit on different volumes. 


There is also this related PR (https://github.com/h2oai/h2o-llmstudio/pull/212) suggesting to use the hosts `.cache` folder.

Reply:
> people may use the same ENV variable for both folders and that may lead to very unexpected results

Ah I see, eliminate the possibility of human-error, don't want to end up with:

```yaml
output_folder: /workspace
data_folder: /workspace
```

by mistake.

Reply:
> So, we can replace these two lines with a single volume that is attached to the Docker image

Given https://github.com/h2oai/h2o-llmstudio/blob/main/Dockerfile#L28-L37 I think the `/workspace` directory is not clean / contains **_more_** than only `data/` and `output/`, which would be deleted by the volume:

```bash
    -v `pwd`:/workspace\
```

- `WORKDIR` and therefore `pwd` is `/workspace`: https://github.com/h2oai/h2o-llmstudio/blob/main/Dockerfile#L27C1-L27C1

Therefore,

```bash
    -v `pwd`/data:/workspace/data \
    -v `pwd`/output:/workspace/output \
```

is still appropriate, ✅  no change required.

> Ultimately, I would probably adapt the basic Docker approach to use a single mounted volume.

This could be changed in another PR? I'd want to avoid breaking changes with this PR and also see if there are other improvements (e.g. #212 ) to be considered before improving the Docker approach.

Reply:
- Pushed change based on [your proposal above](https://github.com/h2oai/h2o-llmstudio/pull/217#discussion_r1251499947) which is less human error prone
- Updated to latest `main`

Reply:
- Pushed fix to Black formatting errors

Reply:
Any other changes requested? 😃